src/microbe_stage/MicrobeAI.cs

```diff
@@ -9,385 +10,179 @@
+                SetMoveSpeed(0.0f);
             }
         }
 
-        // Run reflexes
-        DoReflexes();
-
         // Clear the absorbed compounds for run and rumble
         microbe.TotalAbsorbedCompounds.Clear();
     }
 
-    // There are cases when we want either ||, so here's two state rolls
-    private static bool RollCheck(float ourStat, float dc, Random random)
+    /// <summary>
+    ///   Anything held between calls of the think() method has a chance of having been disposed elsewhere.
+    ///   This sets anything disposed to null to prevent errors. This can probably be removed when issue 2029 is fixed
+    /// </summary>
+    private void ClearDisposedReferences(MicrobeAICommonData data)
     {
-        return random.Next(0.0f, dc) <= ourStat;
+        if (!data.AllMicrobes.Contains(focusedPrey))
+        {
+            focusedPrey = null;
+        }
     }
 
-    private static bool RollReverseCheck(float ourStat, float dc, Random random)
+    private FloatingChunk GetNearestChunkItem(List<FloatingChunk> allChunks, List<Microbe> allMicrobes, Random random)
     {
-        return ourStat <= random.Next(0.0f, dc);
-    }
+        FloatingChunk chosenChunk = null;
 
-    private void DoReflexes()
-    {
-        // For times when its best to tell the microbe directly what to do (Life threatening, attaching to things etc);
-        /* Check if we are willing to run, and there is a predator nearby, if so, flee for your life
-           If it was ran in evaluate environment, it would only work if the microbe was in the neutral state.
-           because we may need more of these very specific things in the future for things like latching onto rocks */
-        // If you are predating and not being engulfed, don't run away until you switch state (keeps predators chasing
-        // you even when their predators are nearby) Its not a good survival strategy but it makes the game more fun.
-        if (predator != null && (lifeState != LifeState.PREDATING_STATE || microbe.IsBeingEngulfed))
+        // If the microbe cannot absorb, no need for this
+        if (microbe.Membrane.Type.CellWall)
+        {
+            return null;
+        }
+
+        // Retrieve nearest potential chunk
+        foreach (var chunk in allChunks)
         {
-            try
+            if (microbe.EngulfSize > chunk.Size * Constants.ENGULF_SIZE_RATIO_REQ
+                && (chunk.Translation - microbe.Translation).LengthSquared()
+                <= (20000.0 * SpeciesFocus / Constants.MAX_SPECIES_FOCUS) + 1500.0)
             {
-                if (!predator.Dead)
+                if (chunk.ContainedCompounds.Compounds.Any(x => microbe.Compounds.IsUseful(x.Key)))
                 {
-                    if ((microbe.Translation - predator.Translation).LengthSquared() <=
-                        (2000 + ((predator.HexCount * 8.0f) * 2)))
+                    chunkList.Add(chunk);
```

Comment:
It doesn't look like chunkList is read anywhere, it can be removed, right?

Reply:
looks like it can be